diff --git a/CMakeLists.txt b/CMakeLists.txt
index 991b6fe..aa97042 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,8 +11,10 @@ find_package(ndt_omp_ros2 REQUIRED)
 
 find_package(Eigen3 REQUIRED)
 find_package(PCL REQUIRED)
+find_package(OpenMP REQUIRED)
 
 find_package(pcl_ros REQUIRED)
+find_package(pcl_conversions REQUIRED)
 list(APPEND CMAKE_PREFIX_PATH "/usr/lib/x86_64-linux-gnu/cmake")
 message(STATUS "CMAKE_PREFIX_PATH:" ${CMAKE_PREFIX_PATH})
 message(STATUS "LD_LIBRARY_PATH:" ${LD_LIBRARY_PATH})
@@ -24,16 +26,20 @@ include_directories(
   include
   ${PCL_INCLUDE_DIRS}
   ${EIGEN_INCLUDE_DIR}
+  ../ndt_omp_ros2/include
 )
 
 add_library(
   calibration_lib
   src/lib/calibExRLidar2Imu.cpp
+  ../ndt_omp_ros2/src/pclomp/voxel_grid_covariance_omp.cpp
+  ../ndt_omp_ros2/src/pclomp/ndt_omp.cpp
 )
 
 target_link_libraries(
   calibration_lib
   ${PCL_LIBRARIES}
+  OpenMP::OpenMP_CXX
 )
 
 ament_target_dependencies(
@@ -43,6 +49,7 @@ ament_target_dependencies(
   tf2_ros
   geometry_msgs
   ndt_omp_ros2
+  pcl_conversions
 )
 
 add_executable(
@@ -54,6 +61,7 @@ target_link_libraries(
   lidar_imu_calib
   ${PCL_LIBRARIES}
   calibration_lib
+  OpenMP::OpenMP_CXX
 )
 
 ament_target_dependencies(
@@ -63,6 +71,7 @@ ament_target_dependencies(
   tf2_ros
   geometry_msgs
   ndt_omp_ros2
+  pcl_conversions
 )
 
 install(
