diff --git a/CMakeLists.txt b/CMakeLists.txt
index 18c6e4c..2f1791f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,14 @@ ament_target_dependencies(align
 ############
 
 install(TARGETS 
+  ndt_omp
+  align
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
+)
+
+install(TARGETS
   align
   DESTINATION lib/${PROJECT_NAME}
 )
@@ -97,5 +105,6 @@ install(
 
 ament_export_dependencies(rclcpp)
 ament_export_include_directories(include)
+ament_export_libraries(ndt_omp)
 
 ament_package()
